x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/ExpressionBuilder.java

```diff
@@ -180,6 +182,15 @@ public Expression visitComparison(ComparisonContext ctx) {
         switch (op.getSymbol().getType()) {
             case SqlBaseParser.EQ:
                 return new Equals(loc, left, right);
+            case SqlBaseParser.NULLEQ:
+                // Simplify to IS NULL to avoid special handling in QueryTranslator later on
+                if (right.equals(NULL)) {
```

Comment:
This logic should be optimizer (changing the tree) instead of the parser which should be pretty accurate to the actual query (helps with reporting errors). Further more it would make this apply if one of the expressions becomes `NULL` in folding and such.

Reply:
You're right, I didn't think this through.